atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/ranges/DbKvsGetRanges.java

```diff
@@ -325,35 +326,44 @@ private String getSimpleRowSelectOneQueryOracle(
         return String.format(
                 SIMPLE_ROW_SELECT_ONE_ORACLE_TEMPLATE,
                 DbKvs.internalTableName(tableRef),
-                getPrefixedTableName(tableRef),
+                getPrimaryKeyConstraintName(tableRef),
                 minMax,
                 getPrefixedTableName(tableRef),
                 extraWhere);
     }
 
+    private String getPrimaryKeyConstraintName(TableReference tableRef) {
+        return truncateToMaxOracleLength("pk_" + getPrefixedTableName(tableRef));
+    }
+
+    private String truncateToMaxOracleLength(String constraintName) {
+        return constraintName
+                .substring(0, Math.min(OracleTableNameMapper.ORACLE_MAX_TABLE_NAME_LENGTH, constraintName.length()));
```

Comment:
Yes, thanks for catching this! I will have a config check for the dbType and generate the constraint name accordingly.